keras/src/backend/mlx/core.py

```diff
@@ -89,6 +101,14 @@ def to_scalar_list(x):
 
         return mx.array(to_scalar_list(x), dtype=mlx_dtype)
 
+    if _is_h5py_dataset(x):
+        if h5py is None:
+            raise ImportError(
+                "h5py must be installed in order to load a model."
```

Comment:
Technically this might be raised outside of model loading context (maybe someone is just reading some h5 data). Let's make the error message generic